### PR TITLE
Refactor 29/backend error handling

### LINKED ISSUE #29: rework error handling based on isco example
- //String debugMessage //debugmessage is not a stupid thing, think through
--  maybe rename it to intMessage and extMessage?

// error code in general as enum is redudnat because the message already tells them what they need to know
// think about what you want to share, expect others want to hack your system
//{
//        "type": "about:blank",   redudant
//        "title": "Bad Request", redudant
//        "status": 400, redudant
//        "detail": "First name is mandatory",
//        "instance": "/apis/user/createUser", redudant
//        "exceptionId": "b66dcca7-223c-486a-9f7a-d95f163d443b", // what is the benefit ?
//        "errorCode": "MISSING_FIELD" redudant
//        }

### PR DESCRIPTION
**Pull request by pleszr**
closes #29

### CODE REVIEW

File: motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java [hunk 1]
```diff
@@ -86,11 +86,18 @@ void post_Users_MissingEmail() throws Exception {
             request.put("firstName", validFirstName);
             request.put("lastName", validLastName);
 
-            mockMvc.perform(post("/user-apis/users")
+            String responseJson =  mockMvc.perform(post("/user-apis/users")
                     .contentType(MediaType.APPLICATION_JSON)
                     .content(TestUtils.asJsonString(request))
                     .accept(MediaType.APPLICATION_JSON))
-                    .andExpect(status().isBadRequest());
+                    .andExpect(status().isBadRequest())
```
Comment (cynran) [inline motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java:93]: This already checks if the response is 400. I don't understand why you assert for it later.

File: motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java [hunk 2]
```diff
@@ -86,11 +86,18 @@ void post_Users_MissingEmail() throws Exception {
...
+                    .andReturn().getResponse().getContentAsString();
+
+            Integer httpCodeFromJson = JsonPath.parse(responseJson).read("$.httpCode");
+            assertEquals(
+                    400,
+                    httpCodeFromJson);
+            TestUtils.assertMissingField(responseJson, "email", "Email is mandatory");
```
Comment (cynran) [inline motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java:100]: you could do this inline as part of the mockMvc.perform like this:

.expectBody().jsonPath("$.name").isEqualTo("Yoda");

File: motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java (hunk 1 above)
Comment (pleszr) [reply motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java:93]: the http code is part of the header and the body too, this checks the header, the assertion checks the body.

i think the real question is if its good practice to have it in the body too (its redundant information and generates the possibility of errors and confusion), but somehow I kinda like it being there (all the insurers I built integration always sent it in the body, I think I am used to it). 

ISCO also sends it and that pushed me over the line to do it here too. What is your view?

File: motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java (hunk 2 above)
Comment (pleszr) [reply motivaa-backend/motivaa-app/src/test/java/com/motivaa/boundary/UserControllerTest.java:100]: awesome, thanks for the tip! Will adjust them.

**Pull request merged**